.github/workflows/sonarcloud.yml

```
@@ -0,0 +1,42 @@
+        env:
+          XOOPS_SKIP_PSL_UPDATE: 1
+        run: composer update --prefer-stable --no-interaction --prefer-dist --no-progress
+
+      - name: Ensure build directory exists
+        run: mkdir -p build/logs
+
+      - name: Run tests with coverage
+        run: ./vendor/bin/phpunit --colors=always --coverage-clover build/logs/clover.xml
+
+      - name: SonarCloud Scan
+        if: ${{ secrets.SONAR_TOKEN != '' }}
```

actionlint (1.7.10)

[error] context "secrets" is not allowed here. available contexts are "env", "github", "inputs", "job", "matrix", "needs", "runner", "steps", "strategy", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)